cirq/protocols/apply_unitary.py

```diff
@@ -304,17 +362,27 @@ def apply_unitaries(unitary_values: Iterable[Any],
             and `default` wasn't specified.
     """
     if args is None:
-        args = ApplyUnitaryArgs.default(len(qubits))
+        # Default to 2 for backwards compatibility
+        max_shape_dict = {qubit: 2 for qubit in qubits}
+        unitary_values = tuple(unitary_values)
+        # Iterate over all unitary values to find the maximum qid_shape
+        for op in unitary_values:
+            for d, qubit in zip(qid_shape_protocol.qid_shape(op), op.qubits):
+                max_shape_dict[qubit] = max(d, max_shape_dict[qubit])
```

Comment:
This looks like something that `cirq.qid_shape` should do for you. If you get an iterable of operations, take the max shape in the obvious way.